### PR TITLE
Patchflow that generates architecture diagram

### PR DESCRIPTION
**Pull request by plon-Susk7**
## PR Checklist

- [x] The commit message follows our guidelines: [Code of conduct](https://github.com/patched-codes/patchwork/blob/main/CODE_OF_CONDUCT.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Does this PR introduce a breaking change?
- [x] Include PR in release notes?

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Build /CI
- [ ] Documentation
- [ ] Others

## What is the current behavior?

There's no patchflow to generate architecture diagrams for a repo.

Issue Number: #951

## What is the new behavior?

The newly made GenerateDiagram patchflow generates mermaid markdown that visualizes overall architecture of the Repo

## Other information

### CODE REVIEW

**Review by CTY-git — CHANGES_REQUESTED** [review]
Can you add a CI test again @plon-Susk7 . Thanks.

**Comment by plon-Susk7** [comment]
> Can you add a CI test again @plon-Susk7 . Thanks.

Done :)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -201,6 +201,13 @@ jobs:
           --github_api_key=${{ secrets.SCM_GITHUB_KEY }} \
           --base_path=tests/cicd/generate_docstring \
           --disable_telemetry
+      
+      - name : Generate Diagrams
+        run: |
+          poetry run patchwork GenerateDiagrams --log debug \
```
Comment (CTY-git) [inline .github/workflows/test.yml:207]: The patchflow name is `GenerateDiagram`, not `GenerateDiagrams` as used here.

**Review by CTY-git — CHANGES_REQUESTED** [review]

**Comment by plon-Susk7** [comment]
hi @CTY-git sorry to bug you. Could you please review this PR?

**Review by CTY-git — APPROVED** [review]

**Pull request merged**